### PR TITLE
fix bug in mirrorwriter removal

### PR DESCRIPTION
**Pull request by whyrusleeping**
I remember writing the code for the mirror writer, but i dont remember it being the code that is there right now. And the code thats there right now just looks wrong to me. I'm fairly certain that its breaking, causing a panic, and that panic is getting eaten by something somewhere, leading to a lock not being released, and finally leading to this stack trace: http://sprunge.us/eYba (note the large number of calls to log waiting to acquire the lock for minutes on end, and nothing actually appearing to hold the lock).

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by whyrusleeping** [comment]
alright, i figured that, while i was at it, i would fix #1893 #1874 and #1808 all at the same time.

**Comment by whyrusleeping** [comment]
this PR improves @whyrusleeping's sanity by 25%!

File: thirdparty/pollEndpoint/main.go [hunk 1]
```diff
@@ -23,25 +23,30 @@ var (
...
+var log = logging.Logger("pollEndpoint")
+
 func main() {
 	flag.Parse()
 
 	// extract address from host flag
 	addr, err := ma.NewMultiaddr(*host)
 	if err != nil {
-		log.WithField("err", err).Fatal("NewMultiaddr() failed")
+		log.Error("NewMultiaddr() failed: ", err)
```
Comment (jbenet) [inline thirdparty/pollEndpoint/main.go]: does this set error code to non zero? main returns fine... maybe you want:

```go
log.Panic
```

**Comment by jbenet** [comment]
@whyrusleeping can you break this into two commits? one to change the deps, and one to fix things?

File: thirdparty/pollEndpoint/main.go (hunk 1 above)
Comment (whyrusleeping) [inline thirdparty/pollEndpoint/main.go]: oooh, right.

**Comment by whyrusleeping** [comment]
@jbenet but then we break bisection?

**Comment by whyrusleeping** [comment]
@jbenet split the commit.

**Comment by whyrusleeping** [comment]
@jbenet ping, this PR fixes a bunch of issues. the hanging one i actually ran into today.

**Comment by jbenet** [comment]
LGTM

**Pull request merged**